### PR TITLE
End Position of Format Mistakes

### LINKED ISSUE #78: Output Start and End Position of an Error
Vala-Lint should output both a start and end position of a format error (like other linters do). This way, the errors could be underlined directly in the source code.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/78006252-output-start-and-end-position-of-an-error?utm_campaign=plugin&utm_content=tracker%2F45980444&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F45980444&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pantor**
Fix #78.

### CODE REVIEW

**Comment by jeremypw** [comment]
There is no change to the terminal output (i.e. the end point of the error is not shown in the output).

**Comment by pantor** [comment]
This is currently intentional, as I think the end position is not really important for the console output. What's your opinion about this?

However, it will be necessary for graphical output in code editors (related to [this code plugin issue](https://github.com/elementary/code/issues/697))

**Comment by jeremypw** [comment]
Is it intended that the begin and end indices seem start at "1" not "0"?
I notice that not all the unit tests have both expected indices entered - is there a reason for this?

**Comment by jeremypw** [comment]
Probably best to block adding new checks until this is merged.

**Comment by pantor** [comment]
The unit tests should now be complete.

Regarding the indices, that's just the standard convention of most (all?) editors - the column starts at 1. It's just easier to keep this convention throughout vala-lint.

**Comment by jeremypw** [comment]
OK fair enough. I guess as a programmer I am so used to counting from zero!

File: lib/Checks/NoteCheck.vala [hunk 1]
```diff
@@ -37,19 +37,14 @@ public class ValaLint.Checks.NoteCheck : Check {
...
                         string message = r.text.slice (index + keyword.length + 1, index_end).strip ();
 
-                        /* Get correct position of note */
-                        int line_count = Utils.get_line_count (r.text[0:index]);
-                        int line = r.begin.line + line_count;
-                        int column = Utils.get_column_in_line (r.text, index);
-                        if (line_count == 0) {
-                            column += r.begin.column;
-                        }
+                        print (@"$index_end\n");
```
Comment (jeremypw) [inline lib/Checks/NoteCheck.vala]: This looks like a debugging message that should have been removed?

Comment (pantor) [reply lib/Checks/NoteCheck.vala]: Oh sure, my bad 🙈

**Pull request merged**